src/Sylius/Bundle/CoreBundle/Resources/config/cart_providers.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Paweł Jędrzejewski
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services
+                               http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="sylius.cart_provider.new.decorated" class="Sylius\Component\Core\Cart\Provider\NewCartProvider" decorates="sylius.cart_provider.new">
```

Comment:
Is there any reason why you keep two definition of ``cart_provider.new``? IMO it should replace the ``sylius.cart_provider.new``.